tests/conftest.py

```diff
@@ -183,323 +177,112 @@ def dummy_input_bundle():
+        return env.deploy(abi, bytecode, *args, **kwargs)
 
     return ir_compiler
 
 
-def _get_contract(
-    w3,
-    source_code,
-    optimize,
-    experimental_codegen,
-    output_formats,
-    *args,
-    override_opt_level=None,
-    input_bundle=None,
-    **kwargs,
-):
-    settings = Settings()
-    settings.optimize = override_opt_level or optimize
-    settings.experimental_codegen = experimental_codegen
-    out = compiler.compile_code(
-        source_code,
-        # test that all output formats can get generated
-        output_formats=output_formats,
-        settings=settings,
-        input_bundle=input_bundle,
-        show_gas_estimates=True,  # Enable gas estimates for testing
+@pytest.fixture(scope="module")
+def compiler_settings(optimize, experimental_codegen, evm_version):
+    return Settings(
+        optimize=optimize, evm_version=evm_version, experimental_codegen=experimental_codegen
     )
-    parse_vyper_source(source_code)  # Test grammar.
-    json.dumps(out["metadata"])  # test metadata is json serializable
-    abi = out["abi"]
-    bytecode = out["bytecode"]
-    value = kwargs.pop("value_in_eth", 0) * 10**18  # Handle deploying with an eth value.
-    c = w3.eth.contract(abi=abi, bytecode=bytecode)
-    deploy_transaction = c.constructor(*args)
-    tx_info = {"from": w3.eth.accounts[0], "value": value, "gasPrice": 0}
-    tx_info.update(kwargs)
-    tx_hash = deploy_transaction.transact(tx_info)
-    address = w3.eth.get_transaction_receipt(tx_hash)["contractAddress"]
-    return w3.eth.contract(address, abi=abi, bytecode=bytecode, ContractFactoryClass=VyperContract)
 
 
 @pytest.fixture(scope="module")
-def get_contract(w3, optimize, experimental_codegen, output_formats):
+def get_contract(env, optimize, output_formats, compiler_settings):
     def fn(source_code, *args, **kwargs):
-        return _get_contract(
-            w3, source_code, optimize, experimental_codegen, output_formats, *args, **kwargs
-        )
+        settings = kwargs.pop("compiler_settings", compiler_settings)
+        if "override_opt_level" in kwargs:
+            settings = Settings(
+                **dict(settings.__dict__, optimize=kwargs.pop("override_opt_level"))
+            )
+        return env.deploy_source(source_code, output_formats, settings, *args, **kwargs)
 
     return fn
 
 
-@pytest.fixture
-def get_contract_with_gas_estimation(tester, w3, optimize, experimental_codegen, output_formats):
-    def get_contract_with_gas_estimation(source_code, *args, **kwargs):
-        contract = _get_contract(
-            w3, source_code, optimize, experimental_codegen, output_formats, *args, **kwargs
-        )
-        for abi_ in contract._classic_contract.functions.abi:
-            if abi_["type"] == "function":
-                set_decorator_to_contract_function(w3, tester, contract, source_code, abi_["name"])
-        return contract
+# todo: this should be removed and replaced with get_contract
```

Comment:
let's replace some of these unneeded get_contract* fixtures

Reply:
Done.. with the downside of introducing a lot of line changes